server/paymentProviders/stripe/creditcard.js

```diff
@@ -215,7 +215,7 @@ const createChargeAndTransactions = async (hostStripeAccount, { order, hostStrip
       OrderId: order.id,
       amount: order.totalAmount,
       currency: order.currency,
-      hostCurrency: balanceTransaction.currency,
+      hostCurrency: balanceTransaction.currency?.toUpperCase(),
```

Comment:
This is the root fix, see https://stripe.com/docs/api/balance_transactions/object. In the DB these values were properly recorded with uppercase thanks to the model's setter.

Reply:
Though one. :dart: